Fix identifier highlight in injected languages.

Earthcomputer opened the pull request:
Identifier highlighting was disabled for single-line editors, which accounts for injected languages inside single-line Java string literals. According to git blame, the original intention of this was to disable identifier highlighting in dialog editors, so I switched the check to explicitly check that.

My use case is that I'm developing a plugin which makes use of a `MultiHostInjector` for a DSL spread across multiple Java string literals. It would significantly improve UX if identifier highlighting was possible in this use case. I could not find any clean workaround which doesn't involve modifying IntelliJ in this way. Besides, I consider this a bug and it's nice to get it fixed :)

Before the fix:
![image](https://github.com/JetBrains/intellij-community/assets/13084089/b2532256-2566-4903-a9da-ae0d5b7e5aed)

After the fix:
![image](https://github.com/JetBrains/intellij-community/assets/13084089/d9c022f9-ffd2-41d0-adc8-9b3e79a37965)

BasLeijdekkers commented on platform/lang-impl/src/com/intellij/codeInsight/daemon/impl/IdentifierHighlighterPassFactory.java:
```diff
@@ -23,7 +24,7 @@ public final class IdentifierHighlighterPassFactory {
                                                           @NotNull Editor editor,
                                                           @NotNull TextRange visibleRange) {
-    if (!editor.isOneLineMode() &&
+    if (!((EditorEx) editor).isEmbeddedIntoDialogWrapper() &&
```
Now  multi-line editors contained in a dialog don't get identifier highlighting anymore. Perhaps the old and new check should be combined?

BasLeijdekkers commented:
Hi @Earthcomputer ,
Thank you for your contribution. Could you create a YouTrack issue for this? That way it becomes part of the release notes and such.

Earthcomputer commented:
Hi, sorry for the delay, I've been away. I can take a look this evening.

Earthcomputer commented:
I addressed the review, and made a YouTrack issue: https://youtrack.jetbrains.com/issue/IDEA-346533/Identifier-highlighting-doesnt-work-in-single-line-language-injections

BasLeijdekkers commented:
Hello, sorry for the delay. Could you squash the two commits into one? That makes it easier to apply the changes to our internal repository.

Earthcomputer commented:
@BasLeijdekkers done

BasLeijdekkers reviewed (approved):
I have applied your fix with some small changes. It should appear on GitHub within a couple of days.
